src/backend/columnar/columnar_customscan.c

```diff
@@ -631,15 +995,29 @@ ColumnarScanPath_PlanCustomPath(PlannerInfo *root,
 	 * CustomScan as the first field. Otherwise, copyObject() will fail to
 	 * copy the additional fields.
 	 */
-	CustomScan *cscan = (CustomScan *) newNode(sizeof(CustomScan), T_CustomScan);
+	CustomScan *cscan = makeNode(CustomScan);
 
 	cscan->methods = &ColumnarScanScanMethods;
 
-	/* Reduce RestrictInfo list to bare expressions; ignore pseudoconstants */
-	clauses = extract_actual_clauses(clauses, false);
+	/* XXX: also need to store projected column list for EXPLAIN */
```

Comment:
right, I hope I can spend some time on #4818 to address your reviews there :)

Reply:
I think that one's not too hard, if it takes more than a couple hours, let me know.

Reply:
I think I made #4818 ready for reviews again, fyi.